### PR TITLE
Add send assessment closing reminder button feature

### PR DESCRIPTION
**Pull request by ekowidianto**
### **Feature**
Added a button to trigger manual email reminder for assessment closing (only applicable to students). This reminder will also be sent to unsubscribed students.

### **UI**
**Student tab**
image

**Staff tab**
image

### **Other changes**
Sending survey closing reminder manually now also includes unsubscribed students.

### CODE REVIEW

**Comment by cysjonathan** [comment]
To make the buttons in submission list page consistent with other MUI buttons.
Can be a separate PR.

**Comment by ekowidianto** [comment]
> To make the buttons in submission list page consistent with other MUI buttons. Can be a separate PR.

Updated in the current PR

File: app/controllers/course/assessment/assessments_controller.rb [hunk 1]
```diff
@@ -209,6 +221,19 @@ def ordered_assessments_by_tab
     tabs.sort_by { |tab_hash| tab_hash[:title] }
   end
 
+  def course_user_ids
+    case params[:course_users]
```
Comment (zhuhanming) [inline app/controllers/course/assessment/assessments_controller.rb:225]: I feel like this method isn't very defensive. Are we expecting `params[:course_users]` to be always specified? What if it contains some garbage value?

File: app/services/course/assessment/reminder_service.rb [hunk 2]
```diff
@@ -57,9 +58,13 @@ def closing_reminder_staff(assessment, students)
   # Returns a Set of students who have not completed the given assessment.
   #
   # @param [Course::Assessment] assessment The assessment to query.
+  # @param [Array<Integer>] course_user_ids course user ids of intended recipient (if specified).
```
Comment (zhuhanming) [inline app/services/course/assessment/reminder_service.rb:61]:
```suggestion
  # @param [Array<Integer>] course_user_ids Course user ids of intended recipients (if specified).
```

File: app/services/course/assessment/reminder_service.rb [hunk 3]
```diff
@@ -57,9 +58,13 @@ def closing_reminder_staff(assessment, students)
...
+  #   If empty, all students will be selected.
+  # @param [Boolean] include_unsubscribed to include unsubscribed students in the reminder (forced reminder)
```
Comment (zhuhanming) [inline app/services/course/assessment/reminder_service.rb:63]:
```suggestion
  # @param [Boolean] include_unsubscribed Whether to include unsubscribed students in the reminder (forced reminder).
```

File: app/services/course/survey/reminder_service.rb [hunk 4]
```diff
@@ -53,8 +53,9 @@ def closing_reminder_staff(survey, students)
   # Returns a Set of students who have not completed the given survey and subscribe to the survey email.
   #
   # @param [Course::Survey] survey The survey to query.
+  # @param [Boolean] include_unsubscribed to include unsubscribed students in the reminder (forced reminder)
```
Comment (zhuhanming) [inline app/services/course/survey/reminder_service.rb:56]:
```suggestion
  # @param [Boolean] include_unsubscribed Whether to include unsubscribed students in the reminder (forced reminder).
```

**Review by zhuhanming — APPROVED** [review]
Amazing work, Eko! Just some minor nits.

File: app/controllers/course/assessment/assessments_controller.rb (hunk 1 above)
Comment (ekowidianto) [reply app/controllers/course/assessment/assessments_controller.rb:225]: Yes we are expecting it to be always specified. Hm.. In that case should we return an error if it's a garbage value?

Comment (zhuhanming) [reply app/controllers/course/assessment/assessments_controller.rb:227]: Would be good to return an error if it's not a value within the array of expected values.

Comment (ekowidianto) [reply app/controllers/course/assessment/assessments_controller.rb:227]: Yup updated!

**Pull request merged**